x-pack/libbeat/common/cloudfoundry/config.go

```diff
@@ -0,0 +1,45 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package cloudfoundry
+
+import (
+	"time"
+
+	"github.com/gofrs/uuid"
+)
+
+type Config struct {
+	// CloudFoundry credentials for retrieving OAuth tokens
+	ClientID     string `config:"client_id" validate:"required"`
+	ClientSecret string `config:"client_secret" validate:"required"`
+
+	// SkipVerify applies to all endpoints
+	SkipVerify bool `config:"skip_verify"`
```

Comment:
how about using `tlscommon.Config` for this? it should standardize wanted params. My understanding is that we could want to support custom certs here, right?

Reply:
I added the support for this using `tls:` to mirror the other configurations that use this. This actually required me to remove the uaago dependency as it didn't allow me to override the TLS config. Looking into that dependency, I determined that it was very simple and the code was only using one method, so I was able to add my own method and use the HTTPClient that has the correct TLS configuration.